### PR TITLE
treat iam 403s as 404

### PR DESCRIPTION
**Pull request by bergman**
It seems Google's IAM API is returning 403's even though the service account used to make the call has ServiceAccountKeyAdmin permission on the service account. Not sure if this is new but haven't seen this problem before.

### CODE REVIEW

**Comment by honnix** [comment]
It's time to have a test case for `ServiceAccountKeyManager` ;)

**Comment by fabriziodemaria** [comment]
Can we add some `todo` comments to remind that we will change this back when Google fixes the problem?

**Comment by fabriziodemaria** [comment]
Following the discussions irl, I think exception handling and related LOGs for the specific Google `Iam` methods should live as much as possible in `ServiceAccountKeyManager` class. But this is not part of this PR anyway.

**Comment by bergman** [comment]
damn codecov… i hit both changed codepaths and the rest of the diff are tests. maybe tests for the tests?

File: styx-scheduler-service/src/main/java/com/spotify/styx/ServiceAccountKeyManager.java [hunk 1]
```diff
@@ -64,7 +64,8 @@ public boolean keyExists(String keyName) throws IOException {
           .execute();
       return true;
     } catch (GoogleJsonResponseException e) {
-      if (e.getStatusCode() == 404) {
+      // TODO: handle 403 correctly once google fixes their API
+      if (e.getStatusCode() == 403 || e.getStatusCode() == 404) {
         return false;
       }
       throw e;
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/ServiceAccountKeyManager.java]: probably this is not covered ;)

**Comment by honnix** [comment]
+1

**Pull request merged**